modules/social_features/social_event/modules/social_event_invite/src/SocialEventInviteBulkHelper.php

```diff
@@ -145,6 +145,114 @@ public static function bulkInviteEmails(array $emails, $nid, array &$context) {
     $context['results'] = $results;
   }
 
+  /**
+   * Send the invites to emails in a batch.
+   *
+   * @param array $users
+   *   Array containing user ids or user emails.
+   * @param string $nid
+   *   The node id.
+   * @param array $context
+   *   The context.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public static function bulkInviteUsersEmails(array $users, $nid, array &$context) {
+    $results = [];
+
+    foreach ($users as $user) {
+      // Todo: Should be merged with extractEmailsFrom from InviteEmailBaseForm.
+      // Remove select2 ID parameter.
+      $user = str_replace('$ID:', '', $user);
+      preg_match_all("/[\._a-zA-Z0-9+-]+@[\._a-zA-Z0-9+-]+/i", $user, $email);
```

Comment:
It would make sense moving this in a different function so it can be unit-tested.

Reply:
<drivebycomment>
This misses any non-ascii character that is valid in an email (e.g. Russian, Asian or French names).

Background: https://stackoverflow.com/a/2049510/576060
Monster regex here: https://stackoverflow.com/a/201378/576060 (please note that I didn't check whether the PHP Regex implementation properly executes that version).
</drivebycomment>